pkg/crc/segment/segment.go

```diff
@@ -0,0 +1,88 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/pborman/uuid"
+	"github.com/segmentio/analytics-go"
+)
+
+type Client struct {
+	segmentClient analytics.Client
+	config        *crcConfig.Config
+}
+
+func NewClient(config *crcConfig.Config) (Client, error) {
+	client, err := analytics.NewWithConfig("cvpHsNcmGCJqVzf6YxrSnVlwFSAZaYtp", analytics.Config{
+		DefaultContext: &analytics.Context{
+			App: analytics.AppInfo{
+				Name:    constants.DefaultName,
+				Version: version.GetCRCVersion(),
+			},
+		},
+	})
+	if err != nil {
+		return Client{}, err
+	}
+	return Client{segmentClient: client, config: config}, nil
+}
+
+func (c Client) Upload(err error) error {
+	defer c.segmentClient.Close()
+	if !c.config.Get(config.ConsentTelemetry).AsBool() {
+		return nil
+	}
+	logging.Info("Uploading the error to segment")
+
+	AnonymousID, uerr := getUserIdentity()
+	if uerr != nil {
+		return uerr
+	}
+
+	logContent, lerr := logging.GetLatestContentOfLogFile(constants.LogFilePath)
+	if lerr != nil {
+		return lerr
+	}
+
+	t := analytics.NewTraits()
+	t = setError(t, err)
+	t = setLogs(t, logContent)
+
+	return c.segmentClient.Enqueue(analytics.Identify{
+		AnonymousId: AnonymousID,
+		Traits:      t,
+	})
+}
+
+func getUserIdentity() (string, error) {
+	if err := os.MkdirAll(constants.TelemetryDirectoryPath, 0750); err != nil {
+		return "", err
+	}
+	if _, err := os.Stat(constants.TelemetryFilePath); !os.IsNotExist(err) {
+		id, err := ioutil.ReadFile(constants.TelemetryFilePath)
+		if err != nil {
+			return "", err
+		}
+		return string(id), nil
```

Comment:
if id is empty, we should regenerate it

Reply:
right... this is a separate process/function. Here we 'detect' it, and act upon the result in the caller. Right?